base/sort.jl

```diff
@@ -1757,6 +1759,41 @@ julia> v
 """
 sort(v::AbstractVector; kws...) = sort!(copymutable(v); kws...)
 
+function sort(x::NTuple;
+              alg::Algorithm=defalg(x),
+              lt=isless,
+              by=identity,
+              rev::Union{Bool,Nothing}=nothing,
+              order::Ordering=Forward,
+              scratch::Union{Vector, Nothing}=nothing)
+    # Can't do this check with type parameters because of https://github.com/JuliaLang/julia/issues/56698
+    scratch === nothing || x isa NTuple{<:Any, eltype(scratch)} || throw(ArgumentError("scratch's eltype can't accommodate all the input elements"))
```

Comment:
```suggestion
    scratch === nothing || (eltype(x) <: eltype(scratch)) || throw(ArgumentError("scratch's eltype can't accommodate all the input elements"))
```

simpler?

Reply:
That would presumably also make the above comment unnecessary.

Reply:
Yes! Using `eltype(x)` is much better.
```suggestion
    scratch === nothing || eltype(x) == eltype(scratch) || throw(ArgumentError("scratch has the wrong eltype"))
```

Reply:
> That would presumably also make the above comment unnecessary.

No, that comment refers to the fact that the best solution would be to define the method as

```julia
function sort(x::NTuple{<:Any,T}; ..., scratch::Union{Vector{T}, Nothing}=nothing) where T
```

But that is prevented by #56698

Reply:
> the best solution would be to define the method as

You could have two methods:
* for the empty `Tuple`, `Tuple{}`
* for nonempty `NTuple`, `Tuple{T, Vararg{T}} where {T}`

Reply:
That would result in code duplication making it worse than the current implementation.
It would also still fail tests due to https://github.com/JuliaTesting/Aqua.jl/issues/86